slang-gfx.h

```diff
@@ -720,6 +722,11 @@ class ISamplerState : public ISlangUnknown
         float                   minLOD          = -FLT_MAX;
         float                   maxLOD          = FLT_MAX;
     };
+
+    /// Returns a native API handle representing this sampler state object.
+    /// When using D3D12, this will be a D3D12_CPU_DESCRIPTOR_HANDLE.
+    /// When using Vulkan, this will be a VkSampler.
+    virtual SLANG_NO_THROW Result SLANG_MCALL getNativeHandle(InteropHandle* outNativeHandle) = 0;
```

Comment:
As a small TODO kind of thing: we probably want our `getNativeHandle` operations to support a second parameter like `const InteropHandleType* requestedType = nullptr`, where the semantics are:

* If `requestedHandleType` is non-null, the back-end will determine if it *can* return a handle of the requested type and do so if it can. It will return an error otherwise.

* If `requestedType` is `nullptr`, the back-end is allowed to pick whatever handle type it prefers to return (of the ones it supports)

This would both let an application proactive ask for just the handle type it expects/needs (rather than needing to validate the type of the returned handle separate from the `getNativeHandle` call itself), and also make sure we are future-proof for cases where more than one handle type may be possible for a single object (e.g., the way our Vulkan device ends up wrapping multiple Vulkan API objects).

Reply:
Agreed.